packages/@lwc/synthetic-shadow/src/polyfills/event/polyfill.ts

```diff
@@ -74,14 +78,24 @@ function patchedTargetGetter(this: Event): EventTarget | null {
 
 function patchedComposedPathValue(this: Event): EventTarget[] {
     const originalTarget = eventTargetGetter.call(this);
-    const originalCurrentTarget = eventCurrentTargetGetter.call(this);
 
     // Account for events with targets that are not instances of Node (e.g., when a readystatechange
     // handler is listening on an instance of XMLHttpRequest).
     if (!(originalTarget instanceof Node)) {
         return [];
     }
 
+    // If the event target is a host element which has a native shadow root instance attached to it,
+    // invoke the native composedPath() method.
+    if (isHostElement(originalTarget)) {
+        const ctorString = (originalTarget as any).shadowRoot?.constructor.toString();
```

Comment:
```suggestion
        const ctorString = originalTarget.shadowRoot!.constructor.toString();
```

Reply:
I don't think we can guarantee that the target has an attached shadow root.